lib/publiccloud/azure.pm

```diff
@@ -283,10 +283,21 @@ sub parse_instance_id
 }
 
 =head2 cleanup
-This method is called called after each test on failure or success to revoke the credentials
+This method is called after each test module to cleanup
 =cut
 
 sub cleanup {
+    my ($self) = @_;
+    $self->SUPER::cleanup();
+    $self->provider_client->cleanup();
+}
+
+
+=head2 destroy
+This method is called called after each test on failure or success to revoke the credentials
```

Comment:
```suggestion
This method is called called after each test run (not module) on failure or success to revoke the credentials
```

I'd make it more explicit, that this is executed once per test run and not once per test module.

Reply:
there is no "revoke of credentials " anywhere in our code

Reply:
called called